Userland/Shell/Builtin.cpp

```diff
@@ -1781,6 +1782,97 @@ ErrorOr<int> Shell::builtin_argsparser_parse(Main::Arguments arguments)
     return 0;
 }
 
+ErrorOr<int> Shell::builtin_read(Main::Arguments arguments)
+{
+    if (!m_in_posix_mode) {
+        raise_error(ShellError::EvaluatedSyntaxError, "read: POSIX builtin used in non-POSIX mode");
+        return 1;
+    }
+
+    bool no_escape = false;
+    Vector<DeprecatedString> variables;
+
+    Core::ArgsParser parser;
+    parser.add_option(no_escape, "Do not interpret backslash escapes", "no-escape", 'r');
+    parser.add_positional_argument(variables, "Variables to read into", "variable", Core::ArgsParser::Required::Yes);
+
+    if (!parser.parse(arguments, Core::ArgsParser::FailureBehavior::Ignore))
+        return 1;
+
+    auto split_by_any_of = " \t\n"_short_string;
+    auto split_by_variable = TRY(lookup_local_variable("IFS"sv));
```

Comment:
pet peeve: `lookup` is  a noun, `look up` is the verb, and functions are named after verbs

Also, this is used only in the if – why not declare it in the if? Then you don't do the local lookup if it's in the env. (Not that it _really_ matters.)